### ISSUE TITLE
tag in string mixins

### ISSUE THREAD

**Issue by do100Han** [pugjs/pug#3087]
I have...
mixin paper(svg, title, title_class, texts, text_class)
  .paper(class != attributes.class)
    if (svg)
      svg.icon.icon_qoute
        use(xlink:href="images/sprite.svg#qoute")
    .paper__content
      if (title)
        h4.title(class = title_class)= title
      if (texts)
        each text in texts
          p.text(class = text_class)= text
options...
+paper(false, false, '', ['*Bla blabla blabla blabla #[a(href="/") blabla]', '*bla bla'], '')
How to make a string appear with a link?

**Comment by do100Han** [comment]
`mixin paper(svg, title, title_class, texts, text_class)
.paper(class != attributes.class)
if (svg)
svg.icon.icon_qoute
use(xlink:href="images/sprite.svg#qoute")
.paper__content
if (title)
h4.title(class = title_class)= title
if (texts)
each text in texts
p.text(class = text_class)= text`

**Comment by algorys** [comment]
I was looking for exactly the same thing and I finally figured out how to do it:

Use following syntax: `tag!= var` or `tag !{var}`

```pug
mixin paper(svg, title, title_class, texts, text_class)
   .paper(class != attributes.class)
     if (svg)
       svg.icon.icon_qoute
         use(xlink:href="images/sprite.svg#qoute")
     .paper__content
       if (title)
         h4.title(class = title_class)= title
       if (texts)
         each text in texts
           p.text(class = text_class) !{text}
```

Then use following syntax:

```pug
+paper(
    false, false, "",
    ['*Bla blabla blabla blabla <a href="/">blabla</a>', '*bla bla'],
    ""
)
```

This will make the expected result:

```html
<div class="paper__content">
    <p class="text">*Bla blabla blabla blabla <a href="/">blabla</a></p>
    <p class="text">*bla bla</p>
</div>
```

**Comment by ForbesLindesay** [comment]
Yes, unfortunately there's no native pug way to say "please treat the following content as pug" when it's a parameter to a mixin. This is because I want to avoid treating "string literals" differently from "user strings". We could possibly add some syntax for it.

Actually, thinking as I go along, this issue has a lot in common with https://github.com/pugjs/babel-plugin-transform-react-pug/issues/88

I wonder if we should consider either having a way to say "this attribute/argument is pug, not JavaScript" or if we should extend JavaScript to add a "pug" type.  The mixin case is extra difficult because we need some way to mark that value as something that should be treated as pug/html even if used in a context that is normally escaped.
